app/models/state_file_base_intake.rb

```diff
@@ -407,12 +412,23 @@ def latest_submission
   end
 
   def increment_failed_attempts
+    # If unlock_in time has passed, unlock the intake. Otherwise, the intakes' failed_attempts will
+    # remain unchanged until the verification code is answered correctly (see: unlock_for_login!)
+    # derived from https://github.com/heartcombo/devise/issues/4679#issuecomment-1453450974
+    if lock_expired?
+      unlock_access!
+    end
+
     super
     if attempts_exceeded? && !access_locked?
       lock_access!
     end
   end
 
+  def unlock_for_login!
+    unlock_access! unless access_locked?
+  end
```

Comment:
this is used exclusively right before we go to `IntakeLoginsController#edit`, after we successfully pass the verification code check from both FYST/Sign In flow. We want to reset the `failed_attempts` to start over from 0 on the SSN screen. If the `locked_at` timestamp was 30+ minutes (which is what `access_locked?` checks), I think we can safely unlock the intake